lib/vmpooler/pool_manager.rb

```diff
@@ -761,12 +799,7 @@ def execute!(maxloop = 0, loop_delay = 1)
         end
 
         $config[:pools].each do |pool|
-          if ! $threads[pool['name']]
-            check_pool(pool)
-          elsif ! $threads[pool['name']].alive?
-            $logger.log('d', "[!] [#{pool['name']}] worker thread died, restarting")
-            check_pool(pool)
-          end
+          evaluate_pool(pool, task_limit, maxloop, loop_delay)
```

Comment:
This pattern will tend to favour servicing pools at the lower end of the array if a `max_loop` is set and quite possibly, higher up pools will never get serviced.

e.g. Consider a 10 pool config (P01 to P10) with task_limit 5

1st run - P01 to P05 will get a slot
     P06-P10 will silently return

P01 - P05 will return quickly (duration less than the 5 * `loop_delay`)

2nd run - P01 to P05 will get a slot
     P06-P10 will silently return

and so on.
---
However if `max_loop` of 0 is set, this turns the pool evaluation into a serial process instead of the current parallel process so VM operations will be delayed.